src/swap/v2/router/swap-router.ts

```diff
@@ -0,0 +1,198 @@
+import algosdk, {ALGORAND_MIN_TX_FEE, getApplicationAddress, Transaction} from "algosdk";
+import AlgodClient from "algosdk/dist/types/src/client/v2/algod/algod";
+
+import {CONTRACT_VERSION} from "../../../contract/constants";
+import {ALGO_ASSET_ID} from "../../../util/asset/assetConstants";
+import {getAssetId, isAlgo} from "../../../util/asset/assetUtils";
+import {SupportedNetwork} from "../../../util/commonTypes";
+import {getValidatorAppID} from "../../../validator";
+import {SwapType} from "../../constants";
+import {
+  FetchSwapRouteQuotesPayload,
+  FetchSwapRouteQuotesResponse,
+  GenerateSwapRouterTxnsParams
+} from "../../types";
+import {
+  V2_SWAP_APP_CALL_ARG_ENCODED,
+  V2_SWAP_APP_CALL_SWAP_TYPE_ARGS_ENCODED,
+  V2_SWAP_ROUTER_APP_ARGS_ENCODED
+} from "../constants";
+import {TINYMAN_ANALYTICS_API_BASE_URLS} from "./constants";
+import {getSwapRouterAppID} from "./util";
+
+export async function generateSwapRouterAssetOptInTransaction({
+  client,
+  routerAppID,
+  assetIDs,
+  accountAddress
+}: {
+  client: AlgodClient;
+  routerAppID: number;
+  assetIDs: number[];
+  accountAddress: string;
+}): Promise<Transaction> {
+  const suggestedParams = await client.getTransactionParams().do();
+  // We need to create a NoOp transaction for each asset we want to opt-in to
+  const assetOptInTxn = algosdk.makeApplicationNoOpTxnFromObject({
+    from: accountAddress,
+    appIndex: routerAppID,
+    appArgs: [V2_SWAP_ROUTER_APP_ARGS_ENCODED.ASSET_OPT_IN],
+    foreignAssets: assetIDs,
+    suggestedParams
+  });
+  // The number of inner transactions is the number of assets we're opting in to
+  const innerTransactionCount = assetIDs.length;
+
+  // The fee for the transaction is the fee for the outer transaction (which is a NoOp)
+  // multiplied by the number of inner transactions
```

Comment:
```suggestion
  /**
   * The opt-in transaction fee should cover the total cost of inner transactions,
   * and the outer transaction (thus the +1)
   */
```
Wdyt? I think this can be a bit more accurate and explanatory,
By the way, AFAIK we don't have a convention for comments, but I use `/* */` for multi-line comments, I find it easier to see it's multiple lines at a glance

Reply:
I find them useful outside the function scope since they are occupying 2 extra lines